pkg/cmd/roachtest/roachtestutil/utils_test.go

```diff
@@ -27,3 +28,57 @@ func TestCmdLogFileName(t *testing.T) {
 		cmdLogFileName(ts, nodes, "./cockroach bla --foo bar"),
 	)
 }
+
+func TestToMarkdownTable(t *testing.T) {
+	tests := []struct {
+		name          string
+		input         [][]string
+		expectedOut   string
+		shouldSucceed bool
+	}{
+		{
+			name: "valid table",
+			input: [][]string{
+				{"Name", "Age", "City"},
+				{"Alice", "30", "New York"},
+				{"Bob", "25", "San Francisco"},
+			},
+			expectedOut: `| Name | Age | City |
+| --- | --- | --- |
+| Alice | 30 | New York |
+| Bob | 25 | San Francisco |
+`,
+			shouldSucceed: true,
+		},
+		{
+			name:          "empty data",
+			input:         [][]string{},
+			expectedOut:   "",
+			shouldSucceed: false,
+		},
+		{
+			name: "row with wrong number of columns",
+			input: [][]string{
+				{"Name", "Age", "City"},
+				{"Alice", "30"},
+			},
+			expectedOut:   "",
+			shouldSucceed: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := ToMarkdownTable(tt.input)
+			if err != nil {
+				if tt.shouldSucceed {
+					t.Errorf("ToMarkdownTable() failed when expected to succeed: error = %v", err)
```

Comment:
Might as well check for the expected err since the above would also output the last row index in case of (column number) mismatch.